### DISCUSSION TITLE
Multiple context via hop

### DISCUSSION THREAD

**Discussion by bruceus** [Natizyskunk/vscode-sftp#181] [Q&A] [answered]
Hi,
is there an option to configure multiple context via one hop?
One file for sending like this:
hopA -> Server1, Server2
Thank you,
Michal

**Comment by Natizyskunk** [comment]
I'm not sure to undesrtand your question. Can you be more precise please ? 🙂

**Reply by manafixer** [reply]
I think he is asking if there's an option to setup multiple servers. 

In my case I wanna setup different FTP, so I can choose which to work on.

There's this option?

**Reply by Natizyskunk** [reply]
@sashachirico for this you can easily set up multiple profiles like [this](https://github.com/Natizyskunk/vscode-sftp#profiles) and then use `SFTP: Set Profile` to switch profile.

**Comment by Natizyskunk** [comment, marked as answer]
> Hi, is there an option to configure multiple context via one hop? One file for sending like this: hopA -> Server1, Server2 Thank you, Michal

@bruceus, I've actually never tested it but I think you can do something like this : 

```json
{
  "name": "targets",
  "remotePath": "/path/in/target",

  // hopA
  "host": "hopAHost",
  "username": "hopAUsername",
  "privateKeyPath": "/Users/localUser/.ssh/id_rsa", // <-- The key file is assumed on the local.

  "hop": [
    {
      "name": "target1",
      "context": "project/build",
      "host": "target1Host",
      "username": "target1Username",
      "privateKeyPath": "/Users/hopUser/.ssh/id_rsa", // <-- The key file is assumed on the hop.
    },
    {
      "name": "target2",
      "context": "project/src",
      "host": "target2Host",
      "username": "target2Username",
      "privateKeyPath": "/Users/hopUser/.ssh/id_rsa", // <-- The key file is assumed on the hop.
    }
  ]
}
```
